drivers/ioexpander/ioe_dummy.c

```diff
@@ -139,32 +142,32 @@ static void sim_interrupt(wdparm_t arg);
  * well be pre-allocated.
  */
 
-static struct sim_dev_s g_ioexpander;
+static struct ioe_dummy_dev_s g_ioexpander;
 
 /* I/O expander vtable */
 
-static const struct ioexpander_ops_s g_sim_ops =
+static const struct ioexpander_ops_s g_ioe_dummy_ops =
 {
-  sim_direction,
-  sim_option,
-  sim_writepin,
-  sim_readpin,
-  sim_readpin
+  ioe_dummy_direction,
```

Comment:
```suggestion
    .direction = ioe_dummy_direction,
```
and same for others?

Reply:
But drivers/motor/foc/foc_dummy.c and many many other drivers use this. Should we fill an issue to clean-up all this and get back to C89?

Reply:
Done for FOC driver. arch driver can relax the requirement since widely used compiler support the new C version, but the common code is better to avoid C99 only feature.

Reply:
The case is that C89 compatibility in such cases cost a lot if there are similar function prototypes but there is a bug in initialization list. Like in drivers/sensors/as5048b.c from https://github.com/apache/incubator-nuttx/pull/5140

Reply:
I really believe that we should use designated initializers here and in other places of common code. NuttX code is far away from C89 already and I do not see way back. Otherwise we need to clean-up all the `inline`, `<stdbool.h>`, `<inttypes.h>`, `__VA_ARGS__` and other C99 features from the code. I do not think that it is doable and we need to try keeping C89 compliance.